weatherAplication/Modules/ShowWeather/View/ShowWeatherViewController.swift

```diff
@@ -13,29 +13,17 @@ class ShowWeatherViewController: UIViewController {
     private let cityLabel = UILabel()
     private let weatherTableView = UITableView()
 
-    private var dateString = String()
-
-    private var dataArray: [WeatherData] = []
     private let apiClient = APIClient.sharedInstance
+    private var viewModel = ShowWeatherViewModel()
 
     init(data: CityInfo) {
         latitude = data.latitude
         longitude = data.longitude
         super.init(nibName: nil, bundle: nil)
+        viewModel.delegate = self
+        viewModel.getWeather(data: data)
```

Comment:
jak wspomnialem wyzej, latitude i longitude tez powinny byc w VM. podsumowujac, zeby to wygladalo super czysto, to `getWeather(data:) `powinno byc wywolywane w inicie ViewModelu, init VC przyjmowac jako parametr tylko VMContract, a podczas nawigacji z jednego VC do drugiego przekazywac CityInfo w srodku ViewModelu